### PR TITLE
Add Frontend tests

### PR DESCRIPTION
**Pull request by phonemkyaw**
Added tests for Frontend

1. Contains one CloudFront distribution
2. Contains one S3 bucket
3. Default behavior of CloudFront as S3 Origin
4. Additional behavior for API Gateway is set to /prod

### CODE REVIEW

File: test/Frontend.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                Origins: Match.arrayWith([
+                    Match.objectLike({
+                        Id: defaultCacheId,
+                        S3OriginConfig: Match.objectLike({})
+                    })
+                ])}
+        });
+    });
+    
+    test('API Gateway path is set to /prod', ()=> {
```
Comment (BrianFarnhill) [inline test/Frontend.test.ts:58]: Can we add a test to make sure that the APIGW origin has caching disabled?

File: test/Frontend.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    test('CloudFront distribution count == 1', () => {
+        assert.resourceCountIs('AWS::CloudFront::Distribution', 1);
+    });
+    
+    test('S3 Bucket count == 1', () => {
+        assert.resourceCountIs('AWS::S3::Bucket', 1);
+    });
+    
+    test('CloudFront default behaviour == S3', () => {
```
Comment (BrianFarnhill) [inline test/Frontend.test.ts]: Can we add a test to make sure that the default caching policy is applied to the S3 origin

**Review by BrianFarnhill — CHANGES_REQUESTED** [review]
Just want a couple of extra tests, other than that looks good.

File: test/Frontend.test.ts (hunk 1 above)
Comment (phonemkyaw) [reply test/Frontend.test.ts:58]: I'm working on it. There'll be a new test for that.

File: test/Frontend.test.ts (hunk 2 above)
Comment (phonemkyaw) [reply test/Frontend.test.ts]: I guess [this](https://github.com/phonemkyaw/Demos.CDKApiPipeline/blob/21ca2e5b459fea1bbb788e2d7e7a850bb5a3de73/test/Frontend.test.ts#L32 ) is the test?

Comment (BrianFarnhill) [reply test/Frontend.test.ts]: That tests that the origin is S3. If you look at the specific properties on it you'll see the cache policy ID of `4135ea2d-6df8-44a3-9df3-4b5a84be39ad` in there - this is the part I'm looking for. Have a look at [the doco for managed cache policies](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/using-managed-cache-policies.html) and you can see the ID's of the default policies so you know where I got that from.

**Comment by phonemkyaw** [comment]
Updated managed cache policies based on the feedback. Please take a look.

**Comment by BrianFarnhill** [comment]
@phonemkyaw Do a pull of whats in my main branch. Your copy of `./jest.config.js` isn't up to date (it's missing the test reporter line, there's also changes to package.json and the package lock too to get the module it uses). Once that's there to build will pass in CodeBuild. It's only failing because it can't find the test reporting files - the build itself is completely though so it's not a logic error in your code.

**Comment by phonemkyaw** [comment]
> @phonemkyaw Do a pull of whats in my main branch. Your copy of `./jest.config.js` isn't up to date (it's missing the test reporter line, there's also changes to package.json and the package lock too to get the module it uses). Once that's there to build will pass in CodeBuild. It's only failing because it can't find the test reporting files - the build itself is completely though so it's not a logic error in your code.

Merged the main branch. Seems like the build is successful. Please take a look.

**Review by BrianFarnhill — APPROVED** [review]

**Pull request merged**